src/buffer4/worker.rs

```diff
@@ -3,6 +3,7 @@ use super::{
     message::Message,
 };
 use futures::stream::StreamExt;
+use std::future::Future;
```

Comment:
i think that if we apply my suggestion on `option_and_then` below, we wouldn't need to import the `Future` trait 
```suggestion
```